pkg/chunk/storage_client.go

```diff
@@ -19,8 +19,8 @@ type StorageClient interface {
 	QueryPages(ctx context.Context, query IndexQuery, callback func(result ReadBatch, lastPage bool) (shouldContinue bool)) error
 
 	// For storing and retrieving chunks.
-	PutChunk(ctx context.Context, key string, data []byte) error
-	GetChunk(ctx context.Context, key string) ([]byte, error)
+	PutChunks(ctx context.Context, chunks []Chunk, keys []string, data [][]byte) error
+	GetChunks(ctx context.Context, chunks []Chunk) ([]Chunk, error)
```

Comment:
The chunks you're passing in are "empty", in that they just describe what to fetch.

I could separate out the parsed chunk ID (`ChunkDescriptor`) from the chunk itself, and embed on in another.  Then it could take a ChunkDescriptor and return a Chunk.  WDYT?